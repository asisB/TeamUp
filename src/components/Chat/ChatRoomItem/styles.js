import { StyleSheet } from "react-native"

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        padding: 10,
    },
    text: {
        color: 'gray',
    },
    image: {
        height: 60,
        width: 60,
        borderRadius: 30,
        marginRight: 10,

    },
    badgeContainer: {
        backgroundColor: 'red',
        width: 20,
        height: 20,
        borderRadius: 10,
        borderWidth: 1,
        borderColor: 'white',
        alignItems: 'center',
        justifyContent: 'center',
        position: 'absolute',
        left: 55,
        top: 10,
    },
    badgeText: {
        color: 'white',
        fontSize: 12,
    },
    rightContainer: {
        flex: 1,
        justifyContent: 'center',
    },
    row: {
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
    name: {
        fontWeight: 'bold',
        fontSize: 18,
        marginBottom: 3,
    }
})

export default styles;
