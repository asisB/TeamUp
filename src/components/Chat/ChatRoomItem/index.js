import React from 'react'
import { View, Text, Image } from 'react-native'
import styles from './styles'

const ChatRoomItem = () => {
    return (
        <View>
            <View style={styles.container}>
                <Image source={{ uri: 'https://notjustdev-dummy.s3.us-east-2.amazonaws.com/avatars/vadim1.JPG' }} style={styles.image} />
                <View style={styles.badgeContainer}>
                    <Text style={styles.badgeText}>4</Text>
                </View>
                <View style={styles.rightContainer}>
                    <View style={styles.row}>
                        <Text style={styles.name}>Elon Musk</Text>
                        <Text style={styles.text}>11:11 AM</Text>
                    </View>
                    <Text numberOfLines={1} style={styles.text}>Hola Hola coca cola</Text>
                </View>
            </View>
        </View>
    )
}

export default ChatRoomItem;