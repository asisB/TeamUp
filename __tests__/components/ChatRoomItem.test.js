import React from 'react';
import renderer from 'react-test-renderer';
import ChatRoomItem from '../../src/components/Chat/ChatRoomItem';  

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({ navigate: jest.fn() }), 
}));

const chatRoom = {
  id: '1',
  users: [
    {
      name: 'John Doe',
      imageUri: 'https://example.com/john.jpg',
    },

  ],
  newMessages: 3,
  lastMessage: {
    content: 'Hello, how are you?',
    createdAt: '2023-09-21T12:00:00Z',
  },
};

test('ChatRoomItem component renders correctly', () => {
  const tree = renderer.create(<ChatRoomItem chatRoom={chatRoom} />).toJSON();

  expect(tree).toMatchSnapshot();
});
