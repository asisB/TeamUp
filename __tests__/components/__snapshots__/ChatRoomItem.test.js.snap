// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatRoomItem component renders correctly 1`] = `null`;
